pkg/job_controller/job.go

```diff
@@ -324,6 +330,108 @@ func (jc *JobController) ReconcileJobs(job interface{}, replicas map[apiv1.Repli
 	return result, nil
 }
 
+func (jc *JobController) CreateCronJob(meta metav1.Object, runtimeObj runtime.Object, policy *apiv1.RunPolicy) error {
+	cronJob := &appv1.Cron{}
+	isCronJob := policy != nil && policy.CronPolicy != nil
+	if err := jc.Client.Get(context.Background(), types.NamespacedName{
+		Namespace: meta.GetNamespace(),
+		Name:      meta.GetName(),
+	}, cronJob); err != nil {
+		if errors.IsNotFound(err) {
+			if !isCronJob {
+				//Do not need to run regularly
+				return nil
+			}
+			instance, err := jc.createCronInstance(policy, runtimeObj)
+			if err != nil {
+				return err
+			}
+			return jc.Client.Create(context.Background(), instance)
+		}
+		return err
+	}
+	if !isCronJob {
+		//Job no longer needs to run regularly, delete the originally created cronJob
+		return jc.Client.Delete(context.Background(), cronJob)
+	}
+	//If cronPolicy or cronTemplate change, the cronJob needs to be updated
+	newCronJob := cronJob.DeepCopy()
+	needUpdate := false
+	if !reflect.DeepEqual(cronJob.Spec.CronPolicy, *policy.CronPolicy) {
```

Comment:
you may compare `newInstance.Spec` with `cronJob.Spec` directly , and two-step `DeepEqual` comparison can be simplified into one.